crates/common/src/input.rs

```diff
@@ -132,6 +104,27 @@ impl InputFile {
     pub fn abs_path(&self, db: &dyn InputDb, ingot: InputIngot) -> Utf8PathBuf {
         ingot.path(db).join(self.path(db))
     }
+
+    pub fn new(db: &dyn InputDb, path: Utf8PathBuf, ingot: InputIngot) -> Self {
+        input_for_file_path(db, FilePath::from(db, path), ingot)
+    }
+}
+
+#[salsa::tracked]
+pub fn input_for_file_path<'db>(
+    db: &'db dyn InputDb,
+    path: FilePath<'db>,
+    ingot: InputIngot,
+) -> InputFile {
+    // Check if the ingot already has a file with the same path
+    for file in ingot.files(db).iter() {
+        if file.path(db).as_path() == path.path(db) {
+            return *file;
+        }
+    }
```

Comment:
This was necessary so that ingots with manually created/added `InputFile`s (right now just the core ingot) will have their preset `InputFile`s returned.

Reply:
(see here: https://github.com/ethereum/fe/pull/1070/files#diff-994bdd66c9bcceebf3b2f96d28bdb3d35c1ec3fe916c943fac1ac855f0fc7b48R24 )